File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.cpp
class Solution {
public:
    int maxDistance(vector<int>& nums1, vector<int>& nums2) {
        int res = 0;
        int mid;
        
        for (int i = 0; i < nums1.size(); ++i) {
            int left = i, right = nums2.size() - 1;
            while (left <= right) {
                mid = left + (right - left) / 2;
                if (nums2[mid] < nums1[i]) right = mid - 1;
                else left = mid + 1;
            }
            res = max(res, right - i);
        }
        return res;
    }
};

File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.py
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        n1, n2 = len(nums1), len(nums2)
        
        
        def binary_search(i):
            left, right = i, n2 - 1
            while left <= right:
                mid = left + (right - left) // 2
                if nums2[mid] < nums1[i]:
                    right = mid - 1
                else:
                    left = mid + 1
            return right
        
        res = 0
        for i in range(n1):
            idx = binary_search(i)
            res = max(res, idx - i)
        return res